### PR TITLE
Add support for aarch64

### PR DESCRIPTION
**Pull request by odidev**
### Motivation

Fixed the tests issue on aarch64. Previously tests was failing on aarch64 with below error: 
`[^[[1;31mERROR^[[m] testCompressDecompress(org.apache.pulsar.common.compression.CompressorCodecBackwardCompatTest)  Time elapsed: 0.02 s  <<< FAILURE!
java.lang.NoClassDefFoundError: Could not initialize class org.xerial.snappy.Snappy
        at org.apache.pulsar.common.compression.CompressionCodecSnappyJNI.decode(CompressionCodecSnappyJNI.java:60)
        at org.apache.pulsar.common.compression.CompressorCodecBackwardCompatTest.testCompressDecompress(CompressorCodecBackwardCompatTest.java:66)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
`

### Modifications

Upgrade netty-tcnative to 2.0.33.Final and snappy-java to 1.1.7.6 which includes aarch64 support.

### CODE REVIEW

**Comment by odidev** [comment]
Updated snappy-java to 1.1.7.5 because snappy-java latest version 1.1.7.6 is having some issues with jdk8.That's why previously unit tests were failing. 
https://github.com/xerial/snappy-java/issues/247
https://github.com/xerial/snappy-java/issues/251

**Comment by odidev** [comment]
Hi,
Please have a look into this PR.

**Comment by shobhitps** [comment]
Hi,

Please have a look into this PR.

It will be useful if arm64 jobs are added in Travis-CI. If you are interested, I can help by raising another PR for adding arm64 jobs in Travis-CI.

**Comment by odidev** [comment]
Hi,

Please have a look into this PR.

It will be useful if arm64 jobs are added in Travis-CI. If you are interested, I can help by raising another PR for adding arm64 jobs in Travis-CI.

File: distribution/server/src/assemble/LICENSE.bin.txt [hunk 1]
```diff
@@ -371,7 +371,8 @@ The Apache Software License, Version 2.0
     - io.netty-netty-transport-native-epoll-4.1.51.Final.jar
     - io.netty-netty-transport-native-unix-common-4.1.51.Final.jar
     - io.netty-netty-transport-native-unix-common-4.1.51.Final-linux-x86_64.jar
-    - io.netty-netty-tcnative-boringssl-static-2.0.30.Final.jar
+    - io.netty-netty-transport-native-kqueue-4.1.51.Final-osx-x86_64.jar
```
Comment (eolivelli) [inline distribution/server/src/assemble/LICENSE.bin.txt:374]: why do we need kqueue ?

**Review by eolivelli — CHANGES_REQUESTED** [review]

File: distribution/server/src/assemble/LICENSE.bin.txt (hunk 1 above)
Comment (odidev) [reply distribution/server/src/assemble/LICENSE.bin.txt]: Agree, that is not required. Changed accordingly. Thanks.

**Review by codelipenghui — APPROVED** [review]

**Comment by odidev** [comment]
Thanks for merging.

It will be useful if arm64 jobs are added to Travis-CI. If you are interested, I can help by raising another PR for adding arm64 jobs in Travis-CI.

**Comment by codelipenghui** [comment]
@odidev After merged this PR, I can't build the source code on my laptop(macos 10.15.7). Here is the problem that I met:

 Run `mvn clean install -DskipTest` on the master branch

Then, the problem appears

```
[INFO] --- exec-maven-plugin:1.6.0:exec (rename-epoll-library) @ pulsar-broker-shaded ---
----- Renaming epoll lib in /Users/lipenghui/GitHub/pulsar/pulsar-broker-shaded/target/pulsar-broker-shaded.jar ------
checkdir error:  /var/folders/t4/w6thygwj0y596w0j5h_pzlw40000gn/T/tmp.A0pwRE4r/META-INF/license exists but is not directory
                 unable to process META-INF/license/.
checkdir error:  /var/folders/t4/w6thygwj0y596w0j5h_pzlw40000gn/T/tmp.A0pwRE4r/META-INF/license exists but is not directory
                 unable to process META-INF/license/LICENSE.tomcat-native.txt.
checkdir error:  /var/folders/t4/w6thygwj0y596w0j5h_pzlw40000gn/T/tmp.A0pwRE4r/META-INF/license exists but is not directory
                 unable to process META-INF/license/LICENSE.boringssl.txt.
checkdir error:  /var/folders/t4/w6thygwj0y596w0j5h_pzlw40000gn/T/tmp.A0pwRE4r/META-INF/license exists but is not directory
                 unable to process META-INF/license/LICENSE.aix-netbsd.txt.
checkdir error:  /var/folders/t4/w6thygwj0y596w0j5h_pzlw40000gn/T/tmp.A0pwRE4r/META-INF/license exists but is not directory
                 unable to process META-INF/license/LICENSE.mvn-wrapper.txt.
[ERROR] Command execution failed.
org.apache.commons.exec.ExecuteException: Process exited with an error: 2 (Exit value: 2)
    at org.apache.commons.exec.DefaultExecutor.executeInternal (DefaultExecutor.java:404)
    at org.apache.commons.exec.DefaultExecutor.execute (DefaultExecutor.java:166)
    at org.codehaus.mojo.exec.ExecMojo.executeCommandLine (ExecMojo.java:804)
    at org.codehaus.mojo.exec.ExecMojo.executeCommandLine (ExecMojo.java:751)
    at org.codehaus.mojo.exec.ExecMojo.execute (ExecMojo.java:313)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)

```

Do you have any idea for fixing this issue? Looks related to the netty related dependency upgrade.

**Comment by codelipenghui** [comment]
Seems only affect the macos, I tried on ubuntu, it works. And after revert the version of netty-tc-native to 2.0.30.Final also works well on macos.

**Pull request merged**